var/spack/repos/builtin/packages/intel-oneapi-compilers/package.py

```diff
@@ -38,35 +37,45 @@ def __init__(self, spec):
 
+    def _join_prefix(self, path):
+        return join_path(self.prefix, 'compiler', 'latest', 'linux', path)
+
+    def _ld_library_path(self):
+        dirs = ['lib',
+                'lib/x64',
+                'lib/emu',
+                'lib/oclfpga/host/linux64/lib',
+                'lib/oclfpga/linux64/lib',
+                'compiler/lib/intel64_lin',
+                'compiler/lib']
+        for dir in dirs:
+            yield self._join_prefix(dir)
+
     def install(self, spec, prefix):
+        # For quick turnaround debugging, comment out line below and
+        # use the copy instead
         super(IntelOneapiCompilers, self).install(spec, prefix)
-        # For quick turnaround debugging, copy instead of install
-        # copytree('/opt/intel/oneapi/compiler', path.join(prefix, 'compiler'),
-        #          symlinks=True)
-        rpath_dirs = ['lib',
-                      'lib/x64',
-                      'lib/emu',
-                      'lib/oclfpga/host/linux64/lib',
-                      'lib/oclfpga/linux64/lib',
-                      'compiler/lib/intel64_lin',
-                      'compiler/lib']
+        # Copy installed compiler instead of running the installer
+        # from shutil import copytree
+        # copytree('/opt/intel/oneapi/compiler', join_path(prefix, 'compiler'),
+        #         symlinks=True)
+
+        rpath = ':'.join(self._ld_library_path())
         patch_dirs = ['compiler/lib/intel64_lin',
                       'compiler/lib/intel64',
                       'bin']
-        eprefix = path.join(prefix, 'compiler', 'latest', 'linux')
-        rpath = ':'.join([path.join(eprefix, c) for c in rpath_dirs])
         for pd in patch_dirs:
-            for file in glob.glob(path.join(eprefix, pd, '*')):
+            for file in glob.glob(self._join_prefix(join_path(pd, '*'))):
                 # Try to patch all files, patchelf will do nothing if
                 # file should not be patched
                 subprocess.call(['patchelf', '--set-rpath', rpath, file])
 
     def setup_run_environment(self, env):
-        env.prepend_path('PATH', join_path(self.prefix,
-                         'compiler', 'latest', 'linux', 'bin'))
-        env.prepend_path('CPATH', join_path(self.prefix,
-                         'compiler', 'latest', 'linux', 'include'))
-        env.prepend_path('LIBRARY_PATH', join_path(self.prefix,
-                         'compiler', 'latest', 'linux', 'lib'))
-        env.prepend_path('LD_LIBRARY_PATH', join_path(self.prefix,
-                         'compiler', 'latest', 'linux', 'lib'))
+        env.prepend_path('PATH', self._join_prefix('bin'))
+        env.prepend_path('CPATH', self._join_prefix('include'))
+        env.prepend_path('LIBRARY_PATH', self._join_prefix('lib'))
+        for dir in self._ld_library_path():
+            env.prepend_path('LD_LIBRARY_PATH', dir)
+        env.set('CC', 'icx')
+        env.set('CXX', 'icpx')
+        env.set('FC', 'ifx')
```

Comment:
The functional changes in this PR appear to be:

* Add all entries of `_ld_library_path()` to `LD_LIBRARY_PATH` rather than just `lib/`
* Set CC and other compiler vars

Is the first one necessary? I mentioned in https://github.com/spack/spack/pull/20717#issuecomment-759155895 that I wouldn't expect that to be required: we should automatically be adding RPATHs to the compiled binaries (since `libimf` is in `required_libs` for the `oneapi` compiler class defined in Spack).

Reply:
The part that sets CC might be useful as an alternative to setting `I_MPI_CC` in https://github.com/spack/spack/pull/20808/files#r564934758

Reply:
`environment_modifications_for_spec` adds `lib/` to `LD_LIBRARY_PATH`, so the choice to make sure the appropriate lib directories are added isn't a big deal. I was curious though if it was strictly necessary. No changes are needed here.

Reply:
It looks good.